### PR TITLE
Shop hookup

### PR DESCRIPTION
**Pull request by matthewfernst**
shop hooked up 👍

### CODE REVIEW

File: client/components/App/Pages/Collection.jsx [hunk 1]
```diff
@@ -64,6 +64,7 @@ const SelectedTab = (props) => {
 };
 
 const PieceSkins = (props) => {
+	console.log(props.data.selfLookup.items)
```
Comment (mrrosoff) [inline client/components/App/Pages/Collection.jsx:67]: Delete for meeee

File: client/components/App/Pages/Collection.jsx [hunk 2]
```diff
@@ -75,6 +76,7 @@ const PieceSkins = (props) => {
 									name={item.name}
 									description={item.description}
 									thumbnail={item.thumbnail}
+									showCheckMark={props.data.selfLookup.items.map(item => item.name).includes(item.name)}
```
Comment (mrrosoff) [inline client/components/App/Pages/Collection.jsx:79]: Plz auto format :)

File: client/components/App/Pages/Shop.jsx [hunk 3]
```diff
@@ -58,6 +78,7 @@ const SelectedTab = (props) => {
 };
 
 const PieceSkins = (props) => {
+	const [confirmPurchase, setConfirmPurchase] = useState(false);
```
Comment (mrrosoff) [inline client/components/App/Pages/Shop.jsx:81]: Ok but what if I click one but there are many, does it show the new UI for all of them?

File: client/graphql/mutation.js [hunk 4]
```diff
@@ -209,3 +209,17 @@ export const SendMessage = gql`
...
+
+export const BuyItem = gql`
+	mutation BuyItem($itemId: ID!){
+		buyItem(itemId: $itemId){
+			username
+			money
+			items{
+				type
+				name
+				description
```
Comment (mrrosoff) [inline client/graphql/mutation.js:221]: Add thumbnail here!

File: server/gqlTypes/mutation.js [hunk 5]
```diff
@@ -470,6 +470,34 @@ const checkFriends = async (context, friendUsername) => {
...
+			{ _id: context.userId },
+			{ $addToSet: { items: ObjectId(args.itemId) } },
+			{ $set: { money: userRecord.money - itemRecord.cost } },
+			{ returnOriginal: false }
+		);
+
+	const modifiedUser = await context.db
+		.collection("Users")
+		.findOneAndUpdate(
+			{ _id: context.userId },
```
Comment (mrrosoff) [inline server/gqlTypes/mutation.js:493]: You are subtracting money twice, why?

**Review by mrrosoff — CHANGES_REQUESTED** [review]
Nice, couple tweaks needed

**Comment by mrrosoff** [comment]
Closes #4

File: client/components/App/Pages/Shop.jsx (hunk 3 above)
Comment (matthewfernst) [reply client/components/App/Pages/Shop.jsx]: fixed in recent push

**Review by mrrosoff — CHANGES_REQUESTED** [review]

**Review by mrrosoff — APPROVED** [review]

**Pull request merged**